### DISCUSSION TITLE
Virtual Serial/TTY

### DISCUSSION THREAD

**Discussion by spidgrou** [Louisvdw/dbus-serialbattery#1023] [General]
Hello everybody, and thanks for this incredible driver. really thanks.

I built a serial interface with esp8266 and esp-link to see my JBD bms via TCP on my LAN.
So for example if I use the windows software HW Virtual Serial Port I can associate a COM port to the IP address of my JBD bms, it is exactly the same as having with a cable.

Is it possible setup the driver to connect to a BMS with a IP Address?
Or how can I create a virtual TTY to connect to my BMS's IP Address?

Thanks
Ciao

**Comment by mdkeil** [comment]
> Or how can I create a virtual TTY to connect to my BMS's IP Address?

You can take a look at [ser2net](https://github.com/cminyard/ser2net). I used that on my Raspberry to link an ir-receiver (/dev/ttyAMA0) to a ip/port.

**Reply by spidgrou** [reply]
Thanks! 
But I have no idea how to install on Venus OS. 😒
Any help? 
Thank you

**Reply by mdkeil** [reply]
Ah ok.. on VenusOS you can install the large image (for Node-Red) and make the link via serial-Node and TCP-out-Node in Node Red .. but the problem will be, that serialbattery can't connect to this device anymore.
